### DISCUSSION TITLE
Add support for KNXnet/IP secure

### DISCUSSION THREAD

**Discussion by farmio** [XKNX/xknx#840] [Ideas]
Xknx is currently missing support for KNXnet/IP secure. 
It would be great if someone could tackle the severe task of adding this.

Other libraries that have implemented this for reference:
https://github.com/calimero-project/calimero-server/commit/509a40231a83724d5ad635bdb3450704376363d8
https://github.com/acaprojects/ruby-knx/commit/9598573b3acdbacbaa9372a549f9eaa8f07ab6ad

**Comment by millerf** [comment]
I would love to have this feature to be able to test with calimero without having to buy some hardware!

❤️

**Comment by millerf** [comment]
I have been running 2 days without understanding why xknx wasn't finding my calimero-server...

**Comment by DenisMir** [comment]
Anything new in this area?

**Reply by farmio** [reply]
I'm currently working on implementing TCP tunneling - which would be one requirement for secure tunneling. 

Let's see. 
We still don't even have secure ready hardware to test anything against 🤷

**Reply by farmio** [reply]
Apart from that there is #832 which would (hopefully) enable to use Calimero as secure proxy for xknx (HA).

**Reply by farmio** [reply]
Well, theres #875 now for a start. Let's see if this leads to something working.

**Comment by scholzi100** [comment]
> Anything new in this area? 

Have a look at c8dc4695cd26743542ec1df73ed2dac863ca459c
